backend/infrahub/core/branch/tasks.py

```diff
@@ -199,3 +200,19 @@ async def merge_branch(branch: str, conflict_resolution: dict[str, bool] | None
         meta=Meta(initiator_id=WORKER_IDENTITY, request_id=request_id),
     )
     await service.send(message=message)
+
+
+@flow(name="branch-delete")
+async def delete_branch(branch: str) -> None:
+    service = services.service
+
+    await add_branch_tag(branch_name=branch)
+
+    obj = await Branch.get_by_name(db=service.database, name=str(branch))
+    assert obj.id
+    event = BranchDeleteEvent(branch=branch, branch_id=obj.id, sync_with_git=obj.sync_with_git)
+    await obj.delete(db=service.database)
+
+    await service.workflow.submit_workflow(workflow=BRANCH_CANCEL_PROPOSED_CHANGES, parameters={"branch_name": branch})
+
+    await service.event.send(event=event)
```

Comment:
I'd suggest rewriting this a bit. The use of "assert" is generally recommended against in production code as it can behave differently depending on the setup (https://docs.astral.sh/ruff/rules/assert/).

I'm wondering when we'd run into this error if the call to delete succeeded but the call to service.workflow or service.event failed so that the task was retried and the second time we wouldn't be able to get the branch since it was already deleted and then not submit it to the workflow.

Perhaps also move `BRANCH_CANCEL_PROPOSED_CHANGES` to the `BranchDeleteEvent` that already triggers the refresh of branches?

Reply:
If the assert part was just for type-hinting I'd suggest that we add a .get_id() method to StandardNode just like we have on Node.

Reply:
It was for typing, I added get_id on all `StandardNode`